### PR TITLE
add stack exchange

### PR DESCRIPTION
**Pull request by richarddwang**
stack exchange is part of EleutherAI/The Pile, but AFAIK, The Pile dataset blend all sub datasets together thus we are not able to use just one of its sub dataset from The Pile data. So I create an independent dataset using The Pile preliminary components.

I also change default `timeout` to 100 seconds instead of 10 seconds, otherwise I keep getting read time out when downloading source data of stack exchange and cc100 dataset.

When I was creating dataset card. I found there is room for creating / editing dataset card. I've made it an issue. #2797

Also I am wondering whether the import of The Pile dataset is actively undertaken (because I may need it recently)? #1675

### CODE REVIEW

File: datasets/stack_exchange/README.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+- found
+languages:
+- en
+licenses:
+- mit
+multilinguality:
+- monolingual
+pretty_name: Stack Exchange
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:14]:
```suggestion
- 1M<n<10M
```

File: datasets/stack_exchange/README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    - [Dataset Curators](#dataset-curators)
+    - [Licensing Information](#licensing-information)
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://github.com/EleutherAI/stackexchange-dataset
+- **Repository:** [Needs More Information]
+- **Paper:** https://arxiv.org/abs/2101.00027
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:60]:
```suggestion
- **Homepage:** [GitHub](https://github.com/EleutherAI/stackexchange-dataset)
- **Repository:** [Needs More Information]
- **Paper:** [arXiv](https://arxiv.org/abs/2101.00027)
```

File: datasets/stack_exchange/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+### Dataset Summary
+
+This dataset is part of EleutherAI/The Pile dataset and is a dataset for Language Models from processing stackexchange data dump, which is an anonymized dump of all user-contributed content on the Stack Exchange network.
+
+|download_size|34.28 Gib|
+|dataset_size|10.3 Gib|
+
+### Supported Tasks and Leaderboards
+
+- `lm`
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:73]:
```suggestion
The dataset is used for Language Modeling.
```

File: datasets/stack_exchange/README.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+### Languages
+
+- `en`
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:77]:
```suggestion
The dataset is in English.
```

File: src/datasets/utils/file_utils.py [hunk 5]
```diff
@@ -424,7 +424,7 @@ def ftp_get(url, temp_file, timeout=10.0):
         raise ConnectionError(e)
 
 
-def http_get(url, temp_file, proxies=None, resume_size=0, headers=None, cookies=None, timeout=10.0, max_retries=0):
+def http_get(url, temp_file, proxies=None, resume_size=0, headers=None, cookies=None, timeout=100.0, max_retries=0):
```
Comment (lhoestq) [inline src/datasets/utils/file_utils.py:427]: I'm fine with it

File: datasets/stack_exchange/README.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+### Data Fields
+
+- `domain`
+- `text`
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:91]:
```suggestion
- `domain`: Stack Exchange domain of the sample
- `text`: Text content containing both the question and the answer
```

File: datasets/stack_exchange/README.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+@article{pile,
+    title={The {P}ile: An 800GB Dataset of Diverse Text for Language Modeling},
+    author={Gao, Leo and Biderman, Stella and Black, Sid and Golding, Laurence and Hoppe, Travis and Foster, Charles and Phang, Jason and He, Horace and Thite, Anish and Nabeshima, Noa and Presser, Shawn and Leahy, Connor},
+    journal={arXiv preprint arXiv:2101.00027},
+    year={2020}
+}
+```
+
+### Contributions
+[sdtblck](https://github.com/sdtblck) creates the dataset, and [richarddwang](https://github.com/richarddwang) adds the dataset.
```
Comment (lhoestq) [inline datasets/stack_exchange/README.md:165]:
```suggestion
Thanks to [sdtblck](https://github.com/sdtblck) for creating the dataset.
Thanks to [richarddwang](https://github.com/richarddwang) for adding the dataset.
```

**Review by lhoestq — APPROVED** [review]
Thanks ! This looks all good to me :)

**Comment by lhoestq** [comment]
Hi ! Merging this one since it's all good :)

However I think it would also be better to actually rename it `the_pile_stack_exchange` to make things clearer and to avoid name collisions in the future. I would like to do the same for `books3` as well.

If you don't mind I'll open a PR to do the renaming

**Comment by richarddwang** [comment]
> If you don't mind I'll open a PR to do the renaming

@lhoestq That will be nice !!

**Pull request merged**